api/envoy/config/trace/v3/lightstep.proto

```diff
@@ -40,9 +40,14 @@ message LightstepConfig {
   // The cluster manager cluster that hosts the LightStep collectors.
   string collector_cluster = 1 [(validate.rules).string = {min_len: 1}];
 
-  // File containing the access token to the `LightStep
-  // <https://lightstep.com/>`_ API.
-  string access_token_file = 2 [(validate.rules).string = {min_len: 1}];
+  oneof access_token_setting {
```

Comment:
Will DataSource work?https://github.com/envoyproxy/envoy/blob/5a8bfa20dc3c85ecb61826d122696ecaa75dffa0/api/envoy/config/core/v3/base.proto#L334

https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/source/common/config/datasource.cc#L16

Reply:
I suppose so. There would be an overlap with file data source, but that might be OK?

Reply:
Somewhat tangential, is there a way to make a data source as confidential, so it's not leaked through config dump?